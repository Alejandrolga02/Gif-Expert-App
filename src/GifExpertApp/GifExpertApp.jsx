import { useState } from "react";

import { AddCategory, GifGrid } from "../components";

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(["One Punch"]);

	const onAddCategory = (newCategory) => {
		// Add category
		if (categories.includes(newCategory)) return;

		setCategories([newCategory, ...categories]);
	};

	const deleteCategory = (category) => {
		setCategories((categories) => {
			return categories.filter((item) => item !== category);
		});
	};

	return (
		<>
			<h1>GifExpertApp</h1>

			<AddCategory onNewCategory={onAddCategory} />

			{categories.map((category) => (
				<GifGrid key={category} category={category} deleteCategory={deleteCategory} />
			))}
		</>
	);
};
