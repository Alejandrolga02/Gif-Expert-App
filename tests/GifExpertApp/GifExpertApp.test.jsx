import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../../src/GifExpertApp/GifExpertApp";

describe("tests in <GifExpertApp/>", () => {
	test("should match with snapshot", () => {
		const { container } = render(<GifExpertApp />);
		expect(container).toMatchSnapshot();
	});

	test("btn-delete should delete a category", () => {
		render(<GifExpertApp />);

		const deleteButton = screen.getByLabelText("btn-delete");
		fireEvent.click(deleteButton);

		expect(document.getElementById("btn-delete")).toBeFalsy();
	});

	test("onAddCategory should create a category", () => {
		const newCategory = "Back to the Future"

		render(<GifExpertApp />);

		const inputElement = screen.getByRole("textbox", { name: "input" });
		const formElement = screen.getByRole('form');

		fireEvent.change(inputElement, {
			target: {
				value: newCategory,
			},
		});
		fireEvent.submit(formElement);

		const headingsElements = screen.getAllByRole("heading", {level:3})
		expect(headingsElements.length).toBe(2);

		screen.debug();
	});
});
