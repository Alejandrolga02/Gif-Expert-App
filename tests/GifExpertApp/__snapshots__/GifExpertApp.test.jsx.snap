// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <GifExpertApp/> should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      aria-label="input"
      placeholder="Search Gifs"
      type="text"
      value=""
    />
    <button
      aria-label="button"
      class="btn-search"
      type="submit"
    >
      Search GIFs
    </button>
  </form>
  <div
    class="heading-container"
  >
    <h3>
      One Punch
    </h3>
    <button
      aria-label="btn-delete"
      class="btn-delete"
    >
      Delete
    </button>
  </div>
  <div
    class="card-grid"
  >
    <h2>
      Cargando...
    </h2>
  </div>
</div>
`;
